src/aks-preview/azext_aks_preview/tests/latest/test_decorator.py

```diff
@@ -894,6 +896,96 @@ def test_get_disk_driver_update(self):
             ctx_5.get_storage_profile(), storage_profile
         )
 
+        # azuredisk_csi_version value passed and enable-disk-driver passed
+        ctx_6 = AKSPreviewContext(
+            self.cmd,
+            {
+                "azuredisk_csi_version": "v2",
+            },
+            self.models,
+            decorator_mode=DecoratorMode.UPDATE,
+        )
+        storage_profile = (
+            self.models.ManagedClusterStorageProfile(
+                disk_csi_driver = self.models.ManagedClusterStorageProfileDiskCSIDriver(
+                    enabled = True,
+                    version = "v2",
+                ),
+                file_csi_driver = self.models.ManagedClusterStorageProfileFileCSIDriver(
+                    enabled = True,
+                ),
+                snapshot_controller = self.models.ManagedClusterStorageProfileSnapshotController(
+                    enabled = True,
+                ),
+            )
+        )
+        mc = self.models.ManagedCluster(
+            location="test_location",
+            storage_profile=storage_profile,
+        )
+        ctx_6.attach_mc(mc)
+        self.assertEqual(
+            ctx_6.get_storage_profile(), storage_profile
+        )
```

Comment:
If there is a storage profile in mc, a new object will not be created, so you are comparing the object itself, which is always true.

Reply:
Not passing the storage_profile while creating the cluster